Funcky/Monads/Reader/Reader.Monad.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+
+namespace Funcky.Monads
+{
+    public static partial class ReaderExtensions
+    {
+        public static Reader<TEnvironment, TResult> Select<TEnvironment, TSource, TResult>(
+            this Reader<TEnvironment, TSource> source, Func<TSource, TResult> selector)
+            => source
+                .SelectMany(value => selector(value).Reader<TEnvironment, TResult>(), (_, result) => result);
+
+        public static Reader<TEnvironment, TResult> SelectMany<TEnvironment, TSource, TResult>(
+            this Reader<TEnvironment, TSource> source,
+            Func<TSource, Reader<TEnvironment, TResult>> selector)
+            => source.SelectMany(selector, (_, result) => result);
+
+        public static Reader<TEnvironment, TResult> SelectMany<TEnvironment, TSource, TSelector, TResult>(
+            this Reader<TEnvironment, TSource> source,
+            Func<TSource, Reader<TEnvironment, TSelector>> selector,
+            Func<TSource, TSelector, TResult> resultSelector)
+                => environment
+                    =>
+                    {
+                        var value = source(environment);
+                        return resultSelector(value, selector(value)(environment));
+                    };
+
+        public static Reader<TEnvironment, TSource> Reader<TEnvironment, TSource>(this TSource value)
```

Comment:
Yeah I realized that when I wrote the Tests for the Monad-Laws, but I kept it because it is part of the initial implementation because this is an extension function, and you cannot implement extension functions on a generic class... 

I tried to add it in a more consistent way in: a242dd57d22664994f5ef97fdfe92905e9653b5d